healthcheck/healthcheck.go

```diff
@@ -0,0 +1,81 @@
+package healthcheck
+
+import (
+	"context"
+	"time"
+)
+
+type Checker func(*context.Context) (*Check, error)
+
+type Check struct {
+	Name        string    `json:"name"`
+	Status      string    `json:"status"`
+	StatusCode  int       `json:"status_code"`
+	Message     string    `json:"message"`
+	LastChecked time.Time `json:"last_checked"`
+	LastSuccess time.Time `json:"last_success"`
+	LastFailure time.Time `json:"last_failure"`
+}
+
+type HealthResponse struct {
+	Status    string        `json:"status"`
+	Version   string        `json:"version"`
+	Uptime    time.Duration `json:"uptime"`
+	StartTime time.Time     `json:"start_time"`
+	Checks    []Check       `json:"checks"`
+}
+
+type HealthCheck struct {
+	Clients                  []*Client
+	Version                  string
+	StartTime                time.Time
+	CriticalErrorTimeout     time.Duration
```

Comment:
Could the HelathCheck and HealthResponse be combined to 1 struct? they share some fields, and the fields that shouldn't be output can be hidden with `json:"-"`